chia/daemon/server.py

```diff
@@ -869,6 +906,35 @@ async def stop_plotting(self, request: Dict[str, Any]) -> Dict[str, Any]:
             self.state_changed(service_plotter, self.prepare_plot_state_message(PlotEvent.STATE_CHANGED, id))
             return {"success": False}
 
+    async def get_logfile(self, request: Dict[str, Any]):
+        """
+        Read in up to 25MiB of data from the log file and return the data in the ack
+        Creates a long-running task for reading in updates to the log file
+        """
+        service_name = request["service"]
+
+        log_contents: str = ""
+        response: dict = {"success": False, "service_name": service_name}
+
+        if service_name != service_logs:
+            return response
+
+        try:
+            file_path = path_from_root(self.root_path, self.net_config.get("log_filename", "log/debug.log"))
+
+            fp = open(file_path, "r")
```

Comment:
You can use the idiom

```
with open(file_path, "r") as fp:
    code
```

here

Reply:
I keep the fp open and use it within the _watch task when reading more data from the file - this idiom doesn't seem to work in this case - I get errors about reading from a closed file pointer.

Reply:
I think one way around that issue would be to convert this function into a generator that yields its fp. So you could continue using the with open pattern and benefit from the auto-closing logic, while returning execution to the generator's caller (with the fp). Might require some restructuring though. Adam might have a better idea :)